pkg/api/grpcServer_test.go

```diff
@@ -488,6 +495,42 @@ func TestBrokenStream(t *testing.T) {
 	}
 
 	wg.Wait()
+	// we never hear back...
+}
+
+func TestBadBag(t *testing.T) {
+	ts, err := prepTestState(30010)
+	if err != nil {
+		t.Errorf("unable to prep test state %v", err)
+		return
+	}
+	defer ts.cleanupTestState()
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	// install a failing SendMsg to exercise the failure path
+	ts.s.sendMsg = func(stream grpc.Stream, m proto.Message) error {
+		err = errors.New("nothing good")
+		wg.Done()
+		return err
+	}
+
+	ts.reportBadAttr = true
+
+	stream, err := ts.client.Report(context.Background())
+	if err != nil {
+		t.Errorf("Report failed %v", err)
```

Comment:
Yeah, we have `t.Errorf(...); return` all over the place in our tests; we should swap to using `Fatal`. This requires swapping to `t.Run` in table driven tests, though.